### PR TITLE
Split climatology maps into more subtasks

### PR DESCRIPTION
**Pull request by xylar**
For both ocean and sea-ice climatology maps, there is a subtask for remapping observations and one for making each plot (that is, one per season, comparison grid and set of observations).

SST, SSS and MLD can be plotted on an Antarctic stereographic grid by adding 'antarctic' to the `comparisonGrids` list.

A plotting routine for data on a polar grid has also been added, including support for both index and norm color maps (the latter will be used for melt rates and Antarctic temperatures at various depths).

The climatology map plotting subtask under each component has been split out into its own file, the remapping subtasks are under `mpas_analysis.shared.climatology` and each climatology map task has been split into its own file (e.g. `climatology_map_mld.py`) to make it easier to figure out what is going on.

### CODE REVIEW

**Comment by xylar** [comment]
## Testing

I have tested this on my laptop with the QU240 run (copied from Edison) using both python 2 and 3.

The resulting Antarctic plots look as follows (just a few examples):
mldholtetalleyargo_antarctic_gmpas-qu240_ann_years0002-0005
ssthadoi_antarctic_gmpas-qu240_ann_years0002-0005

**Comment by xylar** [comment]
By default, the Antarctic plots aren't performed but they would be easy to add to any runs where they might be useful (e.g. where we are focused on Antarctic biases).

**Comment by xylar** [comment]
@milenaveneziani, I know you're working on other things anyway.  Maybe hold off on reviewing this PR until I make some further changes.  I'm adding a "don't merge" tag.

**Comment by xylar** [comment]
Okay, so clearly I have some work to do on both documentation and CI...

**Comment by xylar** [comment]
## Further Testing

I reran the QU240 test case with python 3 (this time with the subtask for remapping observations included).

I also made sure pytest passed on my machine with python 3.

I made sure the documentation builds correctly and includes the new tasks.

**Comment by xylar** [comment]
@milenaveneziani, I'm currently running some tests on Edison.  If you don't mind running a test or 2, that would be great.  Let me know if you have any concerns as well.

**Comment by xylar** [comment]
@vanroekel, I'd appreciate any feedback you might have on these changes, given that it affects code you wrote.

**Comment by milenaveneziani** [comment]
sure, I'll run some tests later today.

**Comment by xylar** [comment]
## Still Further Testing

I successfully tested with the QU240 run and both python 2 and python 3 on an Edison login node.

I tried to test on a batch node but had hdf5 problems.  I reverted to hdf5 1.8.18 from hdf5 1.10.x (I think) in my test anaconda environments (`acme/software/anaconda_envs/edison_mpas_analysis_py*`) and am retesting.  A very simple test (opening the Nino 34 observation), was successful with 1.8.18 but not with 1.10.x in an interactive job.

**Comment by milenaveneziani** [comment]
I ran a test yesterday on edison, and it seems to have completed fine. But I seem to understand there have been more changes since yesterday night (MST)?

**Comment by xylar** [comment]
@milenaveneziani, the only important changes since yesterday have been to documentation and unit tests, so your test results should still be valid.

**Comment by milenaveneziani** [comment]
yeah, I thought I had fetched the latest changes, but I guess I was wrong (saw some repeated output statements that I don't seem to see in this version).
anyways, I am re-running, and also going through the code. All looks great so far.

File: mpas_analysis/ocean/plot_climatology_map_subtask.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+        Authors
+        -------
+        Luke Van Roekel, Xylar Asay-Davis, Milena Veneziani
+        """
+
+        season = self.season
+        comparisonGridName = self.comparisonGridName
+        self.logger.info("\nPlotting 2-d maps of {} climatologies for {} on "
+                         "the {} grid...".format(self.fieldNameInTitle,
+                                                 season, comparisonGridName))
```
Comment (milenaveneziani) [inline mpas_analysis/ocean/plot_climatology_map_subtask.py]: ah, here is the repeated logging statement (above). It's here and then again below, in `_plot_lat_lon` (not in `_plot_antarctic`). So I guess I would remove it from `_plot_lat_lon`.

**Comment by milenaveneziani** [comment]
@xylar: results of my latest test on edison are here:
http://portal.nersc.gov/project/acme/milena/20171122.beta3rc10_1850.ne30_oECv3_ICG.edison/

all worked well on the login node. I will now do a test in batch.

**Comment by milenaveneziani** [comment]
Test on edison compute nodes worked great as well.
I would just remove the extra logging statement and then merge.

**Comment by xylar** [comment]
Thanks very much @milenaveneziani!  I'll fix the redundant print (thanks for noticing that!) and merge soon.

**Pull request merged**